### PR TITLE
chore: prepare v6.0.0

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description

This PR prepares the codebase for the upcoming `v6.0.0` release. The changes included are: 

1. Bumped the go module from `github.com/desmos-labs/desmos/v5` to `github.com/desmos-labs/desmos/v6`.
2. Added missing changeset entries, and fixed the messages on existing ones.
3. Updated the on-chain upgrade workflow to test the upgrade from `v5.2.0` to `v6.0.0` using the chain state exported from the testnet

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [x] manually tested (if applicable)

### CODE REVIEW

File: x/supply/module.go [hunk 1]
```diff
@@ -180,7 +179,6 @@ type ModuleInputs struct {
 
 	Config *modulev1.Module
 	Cdc    codec.Codec
-	Key    *storetypes.KVStoreKey
```
Comment (RiccardoM) [inline x/supply/module.go:183]: The `x/supply` module does not require any store key

File: app/upgrades/v6/upgrade.go [hunk 2]
```diff
@@ -72,5 +73,9 @@ func (u *Upgrade) Handler() upgradetypes.UpgradeHandler {
 
 // StoreUpgrades implements upgrades.Upgrade
 func (u *Upgrade) StoreUpgrades() *storetypes.StoreUpgrades {
-	return &storetypes.StoreUpgrades{}
+	return &storetypes.StoreUpgrades{
+		Added: []string{
+			tokenfactorytypes.StoreKey,
```
Comment (RiccardoM) [inline app/upgrades/v6/upgrade.go:78]: This is needed since the module is newly added with this upgrade

File: app/upgrades.go [hunk 3]
```diff
@@ -3,12 +3,12 @@ package app
 import (
 	v500 "github.com/desmos-labs/desmos/v5/app/upgrades/v500"
 	v520 "github.com/desmos-labs/desmos/v5/app/upgrades/v520"
-	v530 "github.com/desmos-labs/desmos/v5/app/upgrades/v530"
+	v6 "github.com/desmos-labs/desmos/v5/app/upgrades/v6"
```
Comment (RiccardoM) [inline app/upgrades.go]: We're going to skip the v5.3.0 upgrade and go directly to v6 instead

File: contrib/upgrade_testnet/setup_genesis.py [hunk 4]
```diff
@@ -44,9 +44,9 @@
     genesis['app_state']['auth']['accounts'] += chain_state['app_state']['auth']['accounts']
 
     # x/gov
-    genesis['app_state']['gov']['deposit_params']['max_deposit_period'] = '120s'
-    genesis['app_state']['gov']['voting_params']['voting_period'] = '120s'
-    genesis['app_state']['gov']['deposit_params']['min_deposit'] = [{'amount': '10000000', 'denom': 'udaric'}]
+    genesis['app_state']['gov']['params']['max_deposit_period'] = '120s'
+    genesis['app_state']['gov']['params']['voting_period'] = '120s'
+    genesis['app_state']['gov']['params']['min_deposit'] = [{'amount': '10000000', 'denom': 'udaric'}]
```
Comment (RiccardoM) [inline contrib/upgrade_testnet/setup_genesis.py:49]: The `x/gov` module now uses the `params` object instead of the various `x_params` fields

File: contrib/upgrade_testnet/setup_genesis.py [hunk 5]
```diff
@@ -48,8 +48,15 @@
     genesis['app_state']['gov']['params']['voting_period'] = '120s'
     genesis['app_state']['gov']['params']['min_deposit'] = [{'amount': '10000000', 'denom': 'udaric'}]
 
+    # x/ibc
+    allowed_clients = chain_state['app_state']['ibc']['client_genesis']['params']['allowed_clients']
+    if '09-localhost' not in allowed_clients:
+        allowed_clients.append('09-localhost')
+    genesis['app_state']['ibc']['client_genesis']['params']['allowed_clients'] = allowed_clients
```
Comment (RiccardoM) [inline contrib/upgrade_testnet/setup_genesis.py:55]: This is required otherwise the chain won't start locally. The `09-localhost` is also inside the [default allowed clients list](https://github.com/cosmos/ibc-go/blob/0dbd3f811928b216418a45ea164d184eec86cc67/modules/core/02-client/types/params.go#L12). I believe it wasn't inside previous versions.

File: contrib/upgrade_testnet/submit_upgrade_proposal.sh [hunk 6]
```diff
@@ -52,17 +52,18 @@ for ((i = 1; i < $NODES; i++)); do
...
 
 echo ""
 echo "===> Submitting upgrade proposal"
-RESULT=$($DESMOS tx gov submit-proposal \
+RESULT=$($DESMOS tx gov submit-legacy-proposal \
   software-upgrade $UPGRADE_NAME \
   --title Upgrade \
   --description Description \
   --upgrade-height $UPGRADE_HEIGHT \
+  --no-validate \
```
Comment (RiccardoM) [inline contrib/upgrade_testnet/submit_upgrade_proposal.sh:66]: These changes are needed to make sure the proposal gets submitted properly

File: contrib/upgrade_testnet/submit_upgrade_proposal.sh [hunk 7]
```diff
@@ -52,17 +52,18 @@ for ((i = 1; i < $NODES; i++)); do
   echo $NODE_SECRET | $DESMOS keys add "node$i" --recover --keyring-backend test >/dev/null 2>&1
 done
 
-sleep $SLEEP
+sleep 6s
```
Comment (RiccardoM) [inline contrib/upgrade_testnet/submit_upgrade_proposal.sh:55]: There's no need for 30s sleep time here

File: contrib/upgrade_testnet/submit_upgrade_proposal.sh [hunk 8]
```diff
@@ -79,7 +80,7 @@ sleep 6s
 
 echo ""
 echo "===> Getting proposal id"
-PROPOSAL_ID=$($DESMOS q tx $TX_HASH --node $NODE --output json 2>&1 | jq .logs[0].events[4].attributes[0].value -r)
+PROPOSAL_ID=$($DESMOS q tx $TX_HASH --node $NODE --output json 2>&1 | jq .logs[0].events[1].attributes[0].value -r)
```
Comment (RiccardoM) [inline contrib/upgrade_testnet/submit_upgrade_proposal.sh:83]: The proposal id has changed place inside the returned data

File: x/supply/types/keys.go [hunk 9]
```diff
@@ -1,11 +1,6 @@
...
 
 const (
-	ModuleName = "supply"
-	RouterKey  = ModuleName
-
+	ModuleName   = "supply"
 	QuerierRoute = ModuleName
-
-	QueryCirculatingSupply = "circulatingSupply"
-	QueryTotalSupply       = "totalSupply"
```
Comment (RiccardoM) [inline x/supply/types/keys.go:10]: The removed keys where unused

File: contrib/upgrade_testnet/setup_genesis.py (hunk 5 above)
Comment (dadamu) [reply contrib/upgrade_testnet/setup_genesis.py:55]: I checked the IBC migration document, the localhost client should be upgraded inside the handler manually, we missed it.
https://ibc.cosmos.network/main/migrations/v7-to-v7_1.html

**Comment by dadamu** [comment]
Test on chain upgrade CI is failed because of timetout. Tested manually and it works properly. I think we should increase the timeout time to 1 hour or longer.

**Comment by RiccardoM** [comment]
@dadamu I have no idea why the workflow is failing. I have incremented this to 1 hour but it still has problems. I would guess something related to GitHub for some reason. Anyway, I would like to merge this if you approve it

**Comment by dadamu** [comment]
@RiccardoM Overall looks great, I think it is ready to go after this missing upgrade handler is added:
https://github.com/desmos-labs/desmos/pull/1212#discussion_r1299682669

File: contrib/upgrade_testnet/setup_genesis.py (hunk 5 above)
Comment (RiccardoM) [reply contrib/upgrade_testnet/setup_genesis.py:55]: Thanks @dadamu, I have now merged your code into this pranch with #1216

**Comment by RiccardoM** [comment]
@dadamu Thank, I missed that comment. I have now merged your code using #1216 :pray:

**Review by dadamu — APPROVED** [review]
